velox/exec/tests/utils/QueryAssertions.cpp

```diff
@@ -1134,6 +1158,8 @@ void assertResultsOrdered(
   }
 
   auto expectedRows = duckDbQueryRunner.executeOrdered(duckDbSql, resultType);
+  const auto type =
```

Comment:
In `variant::toJson(TypePtr& type)` function, we are looking for the `kind_`. For example, the `kind_ == BIGINT` for a row, but the type passed to this function from QueryAssertions will be `type == ROW` as it is RowVectorPtr type. This causes a mismatch is the `switch-case` as conditions don't meet and the outputs are wrong.

Reply:
Attaching a screenshot from debugger for reference. Looking at the left-hand side of this screenshot might give much clarity. Not sure if we should change something here.
<img width="1791" alt="Screenshot 2023-08-21 at 9 22 20 PM" src="https://github.com/facebookincubator/velox/assets/136469600/c71b054b-bf77-4b39-84ca-5aa80e927b5d">

Reply:
@majetideepak - Should we write code in `variant::toJson(TypePtr& type)` to compare the `kind_` and `type` and do the following steps in here? If there isn't much impact, I can do that. Not sure if it will impact some other modules here.

Reply:
Fixed. Added some new code in `variant::toJson()` to handle this.